packages/effect/src/MutableList.ts

```diff
@@ -76,13 +76,19 @@ interface MutableListImpl<A> extends MutableList<A> {
 }
 
 /** @internal */
-class LinkedListNode<T> {
-  removed = false
-  prev: LinkedListNode<T> | undefined = undefined
-  next: LinkedListNode<T> | undefined = undefined
-  constructor(readonly value: T) {}
+interface LinkedListNode<T> {
+  removed: boolean
+  value: T
+  prev?: LinkedListNode<T> | undefined
```

Comment:
I'd suggest making these `prev` and `next` values required, though still `| undefined`, to avoid assigning these fields in different orders over time, leading to de-optimizations

Reply:
@TylorS Thanks for the suggestion. Can you clarify that for me though? How would assigning it to `undefined` just to preserve assignment order change the performance characteristics?

Reply:
In the engines, they optimize object creation through the use of "hidden classes", and they utilize order to improve the speculation of inlining optimizations. I find this blog a pretty good primer on object poly/megamorphism (also about a tool that helps spot these) https://devblogs.microsoft.com/typescript/introducing-deopt-explorer/

Reply:
TIL, thanks!

Reply:
No problem! I kinda think of the object's memory being a lot like that of an `Array`, it's optimal to know that property "foo" is always at index 0, and not sometimes at index 1 (or 2 or whatever). The hidden classes essentially map your user-land keys, like "foo", into spaces in memory at particular indexes with other information like the amount of memory that it will likely need allocated